codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java

```diff
@@ -63,10 +63,13 @@ protected void serializeInputDocument(GenerationContext context, OperationShape
         GoWriter writer = context.getWriter();
         StructureShape input = ProtocolUtils.expectInput(context.getModel(), operation);
         String functionName = ProtocolGenerator.getDocumentSerializerFunctionName(input, getProtocolName());
-        writer.addUseImports(SmithyGoDependency.SMITHY_JSON);
 
-        // If there are no members then there's nothing to serialize
-        if (input.members().size() == 0) {
+        if (CodegenUtils.isStubSyntheticClone(ProtocolUtils.expectInput(context.getModel(), operation))) {
+            writer.addUseImports(SmithyGoDependency.STRINGS);
+            writer.openBlock("if request, err = request.SetStream(strings.NewReader(`{}`)); err != nil {",
+                    "}", () -> {
+                        writer.write("return out, metadata, &smithy.SerializationError{Err: err}");
+                    });
```

Comment:
Would this affect rest-json too? Assuming empty json body must always be set to `{}`

Reply:
No this would only be for JSON RPC.